addon/internal-session.js

```diff
@@ -17,25 +17,24 @@ export default Ember.ObjectProxy.extend(Ember.Evented, {
     Ember.assert(`Session#authenticate requires the authenticator to be specified, was "${authenticator}"!`, !Ember.isEmpty(authenticator));
     let theAuthenticator = getOwner(this).lookup(authenticator);
     Ember.assert(`No authenticator for factory "${authenticator}" could be found!`, !Ember.isNone(theAuthenticator));
-    return new Ember.RSVP.Promise((resolve, reject) => {
+    return new RSVP.Promise((resolve, reject) => {
       theAuthenticator.authenticate.apply(theAuthenticator, args).then((content) => {
-        this._setup(authenticator, content, true);
-        resolve();
+        this._setup(authenticator, content, true).then(resolve, reject);
       }, (error) => {
-        this._clear();
-        reject(error);
+        this._clear().finally(() => {
+          reject(error);
+        });
```

Comment:
It's not documented, but if I look at the code it looks like `promise.finally(callback)` is more or less an alias of `promise.then(callback, callback)`:

https://github.com/tildeio/rsvp.js/blob/master/lib/rsvp/promise.js#L482